clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -35,13 +35,12 @@ public InFlightRequests(int maxInFlightRequestsPerConnection) {
     /**
      * Add the given request to the queue for the connection it was directed to
      */
-    public void add(ClientRequest request, long now) {
+    public void add(ClientRequest request) {
         Deque<ClientRequest> reqs = this.requests.get(request.request().destination());
         if (reqs == null) {
-            reqs = new ArrayDeque<ClientRequest>();
+            reqs = new ArrayDeque<>();
             this.requests.put(request.request().destination(), reqs);
         }
-        request.setSendMs(now);
```

Comment:
I agree that we pass the now parameter to InflightRequests.add() currently in doSend(). The reason we added it to InflightRequests is because it is the last point after which a request is ready to be sent over the wire. So we are sure that sendTime is set correctly. I am not strongly against this change though.